### ISSUE TITLE
Issue: reusable workflows not supported

### ISSUE THREAD

**Issue by ljharb** [nektos/act#1114]
## System information

    - Operating System: Mac
    - Architecture: x64
    - Apple M1: no
    - Docker version: 20.10.14
    - Docker image used in `act`: medium?
    - `act` version: act version 0.2.26

## Expected behavior

Run my tests.

## Actual behavior

```sh
$ act
? Please choose the default image you want to use with act:

  - Large size image: +20GB Docker image, includes almost all tools used on GitHub Actions (IMPORTANT: currently only ubuntu-18.04 platform is available)
  - Medium size image: ~500MB, includes only necessary tools to bootstrap actions and aims to be compatible with all actions
  - Micro size image: <200MB, contains only NodeJS required to bootstrap actions, doesn't work with all actions

Default image and other options can be changed manually in ~/.actrc (please refer to https://github.com/nektos/act#configuration for additional information about file structure) Medium
ERRO[0006] failed to obtain container engine info: Error response from daemon: dial unix $HOME/Library/Containers/com.docker.docker/Data/docker.raw.sock: connect: connection refused 
ERRO[0006] 'runs-on' key not defined in Tests: node.js < 10/tests 
ERRO[0006] 'runs-on' key not defined in Tests: node.js < 10/tests 
ERRO[0006] 'runs-on' key not defined in Tests: pretest/posttest/tests 
ERRO[0006] 'runs-on' key not defined in Tests: pretest/posttest/tests 
ERRO[0006] 'runs-on' key not defined in Tests: node.js >= 10/tests 
ERRO[0006] 'runs-on' key not defined in Tests: node.js >= 10/tests 
ERRO[0006] failed to obtain container engine info: Error response from daemon: dial unix $HOME/Library/Containers/com.docker.docker/Data/docker.raw.sock: connect: connection refused 
```

## Workflow and/or repository

https://github.com/inspect-js/has-package-exports/tree/76fd1e7307042d4a2217f2dff72b6c18a39bb9fd/.github/workflows

</details>

## Steps to reproduce

1. Clone the repo
1. check out the `temp` branch, or the SHA above
1. run `act`

## `act` output

See "actual behavior" above.

Basically, I don't need "runs-on" in the main workflow, because the workflow it references already defines it.

**Comment by ChristopherHX** [comment]
Composite Actions, this is supported != reusable workflows, this not

Duplicates #826

**Comment by ljharb** [comment]
ah, thanks for correcting my terminology, and the pointer to the issue.
